spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -318,9 +332,30 @@ protected void doStop() {
 
 		private final AckMode ackMode = getAckMode();
 
+		private final boolean isManualAck = this.ackMode.equals(AckMode.MANUAL);
+
+		private final boolean isManualImmediateAck = this.ackMode.equals(AckMode.MANUAL_IMMEDIATE)
+				|| this.ackMode.equals(AckMode.MANUAL_IMMEDIATE_SYNC);
+
+		private final boolean isAnyManualAck = this.isManualAck || this.isManualImmediateAck;
+
+		private final boolean isRecordAck = this.ackMode.equals(AckMode.RECORD);
+
 		private final boolean syncCommits = KafkaMessageListenerContainer.this.syncCommits;
 
-		private Thread consumerThread;
+		private final long pauseAfter = getPauseAfter();
+
+		private final boolean pauseEnabled = isPauseEnabled();
+
+		private final BlockingQueue<ConsumerRecords<K, V>> recordsToProcess = new LinkedBlockingQueue<>(getQueueDepth());
+
+		private final int queueDepth = getQueueDepth();
+
+		private final BlockingQueue<ConsumerRecord<K, V>> acks = new LinkedBlockingQueue<>();
```

Comment:
Probably not this, since it's the acknowledgment queue for messages already processed, but we might want to do something about fetched messages waiting to process. I think that we do will be highly dependent on the acknowledgment mode. I think that the safest strategy here is to flush/process the remaining acks on `onPartitionRevoked` and gracefully stop the consumer, using `onPartitionsAssigned()` as a trigger to resume with the new partition set and use onPartitionAssigned() as a trigger to start consuming - for consistency, we may want to do that even on a normal start. (If we do that, we will need to have a different lifecycle management for manual assignments, but that may be ok, since manual assignments are not involved in the rebalancing process at all).